### PR TITLE
test: uplift capi versions and templates

### PR DESCRIPTION
**Pull request by rsmitty**
This PR will use the latest templates for v1alpha4 and the supporting
CAPI provider versions. We'll bump again when we land v1beta1

Signed-off-by: Spencer Smith <spencer.smith@talos-systems.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4631)

### CODE REVIEW

File: hack/test/e2e-aws.sh [hunk 1]
```diff
@@ -78,7 +78,7 @@ function setup {
 
   ${CLUSTERCTL} config cluster ${NAME_PREFIX} \
     --kubeconfig /tmp/e2e/docker/kubeconfig \
-    --from https://github.com/talos-systems/cluster-api-templates/blob/v1alpha4/aws/standard/standard.yaml > ${TMP}/cluster.yaml
+    --from https://github.com/rsmitty/cluster-api-templates/blob/main/aws/standard/standard.yaml > ${TMP}/cluster.yaml
```
Comment (rsmitty) [inline hack/test/e2e-aws.sh:81]: will amend once I run everything through testing and get templates merged upstream.

**Review by frezbo — APPROVED** [review]

**Pull request merged**